cns/restserver/util.go

```diff
@@ -89,11 +89,12 @@ func (service *HTTPRestService) restoreState() error {
 		if err == store.ErrKeyNotFound {
 			// Nothing to restore.
 			logger.Printf("[Azure CNS]  No state to restore.\n")
-			return nil
+		} else {
+			logger.Errorf("[Azure CNS]  Failed to restore state, err:%v. Removing azure-cns.json", err)
+			service.store.Remove()
 		}
 
-		logger.Errorf("[Azure CNS]  Failed to restore state, err:%v\n", err)
-		return err
+		return nil
```

Comment:
There is no codepath for this function to return an error now, so we should remove the error return type:
```go
func (service *HTTPRestService) restoreState() {
    ....
}
```